pkg/disttask/framework/scheduler/scheduler_manager.go

```diff
@@ -405,3 +409,94 @@ func (sm *Manager) MockScheduler(task *proto.Task) *BaseScheduler {
 		slotMgr: sm.slotMgr,
 	})
 }
+
+func (sm *Manager) collectMetricsLoop() {
+	logutil.Logger(sm.ctx).Info("collect metrics loop start")
+	ticker := time.NewTicker(defaultCollectMetricsInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-sm.ctx.Done():
+			logutil.BgLogger().Info("collect metrics loop exits")
+			return
+		case <-ticker.C:
+			sm.collectMetrics()
+		}
+	}
+}
+
+func (sm *Manager) collectMetrics() {
+	subtasks, err := sm.taskMgr.GetAllSubtasks(sm.ctx)
+	if err != nil {
+		logutil.BgLogger().Warn("get all subtasks failed", zap.Error(err))
+		return
+	}
+	allNodes, err := sm.taskMgr.GetAllNodes(sm.ctx)
+	if err != nil {
+		logutil.BgLogger().Warn("get all nodes failed", zap.Error(err))
+		return
+	}
+	// taskID => execID => state => cnt
+	subtaskCnt := make(map[int64]map[string]map[proto.SubtaskState]int)
+	taskType := make(map[int64]proto.TaskType)
+	for _, subtask := range subtasks {
+		if _, ok := subtaskCnt[subtask.TaskID]; !ok {
+			subtaskCnt[subtask.TaskID] = make(map[string]map[proto.SubtaskState]int, len(allNodes))
+			for _, node := range allNodes {
+				subtaskCnt[subtask.TaskID][node.ID] = make(map[proto.SubtaskState]int, len(proto.AllSubtaskStates))
+				for _, state := range proto.AllSubtaskStates {
+					subtaskCnt[subtask.TaskID][node.ID][state] = 0
+				}
```

Comment:
```suggestion
```
not needed, default is 0

Reply:
If subtask has been rebalanced, we need reset it.

Reply:
but this is initializing